src/task-actions.ts

```diff
@@ -224,21 +207,15 @@ export async function starknetDeployAction(args: any, hre: HardhatRuntimeEnviron
 }
 
 export async function starknetVoyagerAction(args: any, hre: HardhatRuntimeEnvironment) {
-    let voyagerUrl = VOYAGER_GOERLI_CONTRACT_API_URL;
-
-    if (!isTestnet(args.starknetNetwork)) {
-        if (isMainnet(args.starknetNetwork)) {
-            voyagerUrl = VOYAGER_MAINNET_CONTRACT_API_URL;
-        } else {
-            const msg = `Unknown starknet-network provided: ${args.starknetNetwork}`;
-            throw new HardhatPluginError(PLUGIN_NAME, msg);
-        }
+    const network = getNetwork(args.starknetNetwork, hre, "starknet-network");
+    if (!network.verificationUrl) {
+        throw new HardhatPluginError(PLUGIN_NAME, `Network ${args.starknetNetwork} does not support Voyager verification.`);
```

Comment:
Is there any check for the verification URL? It should be only either testnet or mainnet